patches/tModLoader/Terraria/Achievements/Achievement.cs.patch

```diff
@@ -0,0 +1,53 @@
+--- src/TerrariaNetCore/Terraria/Achievements/Achievement.cs
++++ src/tModLoader/Terraria/Achievements/Achievement.cs
+@@ -1,7 +_,10 @@
+ using System.Collections.Generic;
++using Microsoft.Xna.Framework.Graphics;
+ using Newtonsoft.Json;
+ using Newtonsoft.Json.Linq;
++using ReLogic.Content;
+ using Terraria.Localization;
++using Terraria.ModLoader;
+ using Terraria.Social;
+ 
+ namespace Terraria.Achievements;
+@@ -13,14 +_,18 @@
+ 
+ 	private static int _totalAchievements;
+ 	public readonly string Name;
+-	public readonly LocalizedText FriendlyName;
++	public LocalizedText FriendlyName;
+-	public readonly LocalizedText Description;
++	public LocalizedText Description;
+ 	public readonly int Id = _totalAchievements++;
+ 	private AchievementCategory _category;
+ 	private IAchievementTracker _tracker;
+ 	[JsonProperty("Conditions")]
+ 	private Dictionary<string, AchievementCondition> _conditions = new Dictionary<string, AchievementCondition>();
+ 	private int _completedCount;
++	public int Type;           // TML: Added for keeping track of modded achievements during loading, Native achievements don't get assigned a type as they don't need one.
```

Comment:
Do modded achievements actually need a `Type`? Is there ever any reason to use `AchievementLoader.GetAchievement(int type)`?

Recommend removing